tests/specs/controllers/OpenEAppController.test.js

```diff
@@ -148,4 +156,91 @@ describe('OpenEAppController', () => {
             );
         });
     });
+
+    describe('date countdown', () => {
+        beforeEach(async () => {
+            sandbox.stub(HelperService, 'getUserData').callsFake(() => ({
+                loggedIn: true,
+            }));
+            sandbox
+                .stub(OpenEAppController, '_getApplicationDataFromCasebook')
+                .resolves(resolvedCasebookData);
+        });
+
+        it('returns expired document error if days are below zero', async () => {
+            // when
+            const TWO_DAYS_AFTER_DEADLINE = 1633824000000;
+            sandbox.stub(Application, 'find').resolves(resolvedAppData);
+            sandbox.stub(Date, 'now').callsFake(() => TWO_DAYS_AFTER_DEADLINE);
+            await OpenEAppController.renderPage(reqStub, resStub);
+
+            // then
+            const sailsErrorLogObj = sails.log.error.getCall(0).args[0];
+            expect(sailsErrorLogObj.message === 'Application has expired').to.be
+                .true;
+            expect(resStub.serverError.called).to.be.true;
+        });
+        it('shows correct number of days for 11 day old application', async () => {
+            // when
+            const TWELVE_DAYS_AFTER_COMPLETION = 1630281600000;
+            sandbox.stub(Application, 'find').resolves(resolvedAppData);
+            sandbox
+                .stub(Date, 'now')
+                .callsFake(() => TWELVE_DAYS_AFTER_COMPLETION);
+            await OpenEAppController.renderPage(reqStub, resStub);
+
+            // then
+            expectedPageData.daysLeftToDownload = 9;
+            expect(
+                resStub.view.calledWith(
+                    'eApostilles/openEApp.ejs',
+                    expectedPageData
+                )
+            ).to.be.true;
+        });
+        it('returns error if no createdAt value found', async () => {
+            // when
+            const testAppData = {
+                unique_app_id: 'id_from_apps_table',
+            };
+            sandbox.stub(Application, 'find').resolves(testAppData);
+            sandbox.stub(Date, 'now').callsFake(() => TWO_DAYS_AFTER_COMPLETION);
+            await OpenEAppController.renderPage(reqStub, resStub);
+
+            // then
+            const sailsErrorLogObj = sails.log.error.getCall(0).args[0];
+            expect(sailsErrorLogObj.message === 'No date value found').to.be
+                .true;
+            expect(resStub.serverError.called).to.be.true;
+        });
+    });
+
+    describe('_calculateDaysLeftToDownload', () => {
+        it('returns expected values', () => {
+            // when
+            const TWELVE_DAYS_AFTER_COMPLETION = 1630281600000;
+            const SEVEN_DAYS_AFTER_COMPLETION = 1629849600000;
+            const TWENTY_ONE_DAYS_AFTER_COMPLETION = 1631142000000;
```

Comment:
what happens when you pass 22 days after completion?

Reply:
So it should respond with an error saying the application is expired. The test for that is up here:

https://github.com/UKForeignOffice/loi-application-service/pull/277/files/3d98dd43b1bfc655d0a608f6ffee8caf3d3ad0c3#diff-fef66b6a716dd8342942b557f6df57eaa714e1484126933f96ebd974aa48a980R170

But I'm happy to move it to this section, or move this section above, if you think it makes sense.

Reply:
`calculateDaysLeftToDownload` is more predictable to test (i.e. "pure".. or at least slightly more "pure"). It would be faster (in terms of execution, and writing the tests) to just test that if you're strapped for time. And the render method relies on it. By testing the render function you're having to spy the sails method.

```.js
expect(sailsErrorLogObj.message === 'No date value found').to.be
                 .true;
``` 
is the same message that `calculateDaysLeftToDownload` throws. 

usually if I am testing a render method, I'd be checking if what it renders on a page (or the response body) contains what I'm expecting it to